### PR TITLE
price validation bug fix

### PR DESCRIPTION
**Pull request by nehajain008**
Story [AB#12945](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12945)
Task [AB#13189](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13189)

### CODE REVIEW

**Comment by bjss-chris** [comment]
@nehajain008, are there any additional test cases we can add to cover this scenario?

File: app/helpers/common/priceFormatter.js [hunk 1]
```diff
@@ -9,10 +9,13 @@ export const formatNumber = ({
 
 export const formatPrice = ({ value, minimumFractionDigits = 2, maximumFractionDigits = 4 }) => {
   if (minimumFractionDigits === 0) {
-    if (Number.isInteger(Number.parseFloat(value))) {
-      return formatNumber({ value, maximumFractionDigits });
+    if (!Number.isNaN(Number(value)) && !Number.isNaN(parseFloat(value))) {
```
Comment (bjss-andrew) [inline app/helpers/common/priceFormatter.js:12]: it might work, but the logic here is not easy to follow, would benefit from a refactor

Comment (nehajain008) [reply app/helpers/common/priceFormatter.js]: done few changes as a part of refactoring

**Comment by nehajain008** [comment]
> 
> 
> @nehajain008, are there any additional test cases we can add to cover this scenario?

added testcase for new scenario

**Review by bjss-chris — APPROVED** [review]
I think we might one to revisit the price formatter again at some point as tech debt but I this is definitely an improvement over the previous version.

@bjss-andrew, what do you think?

**Pull request merged**